engines/glk/tads/tads2/qa_scriptor.cpp

```diff
@@ -63,7 +63,7 @@ char *qasgets(char *buf, int bufl) {
 	/* keep going until we find something we like */
 	for (;;)
 	{
-		char c;
+		int c;
```

Comment:
This refers to a character, not an integer. Perhaps `byte` would be a better type in this case?

```suggestion
		byte c;
```

Reply:
@bluegr I used `int` because the `c` variable is used to store the result from `osfgetc()`, whose return value is an `int`:
https://github.com/scummvm/scummvm/blob/083eed99321725786f96d3092131720191beb0ac/engines/glk/tads/os_frob_tads.cpp#L145

So, looking at its name, I thought that this meant to be used as an `fgetc()`-like function, where the result is intentionally stored to an `int` in libc, so that it can always store the value of `EOF` (which is `-1` in practice). As in https://stackoverflow.com/a/49063631. `byte` couldn't guarantee this, since it's unsigned.

...**but** then... after your comment, I had a look at the `osfgetc()` implementation again, and it just calls `readByte()` from `ReadStream`, which says this:
https://github.com/scummvm/scummvm/blob/083eed99321725786f96d3092131720191beb0ac/common/stream.h#L430-L434

i.e. I wonder if `engines/glk/tads/os_frob_tads.cpp` / `engines/glk/tads/tads2/qa_scriptor.cpp` use the interface properly. It checks for an `EOF` but the ReadStream comments tell to check `eos()`. So there's maybe something bigger to fix?

(GLK looks quite fragile at places, in general)

Reply:
It follows the semantics of `fgetc()`, which also returns an int:

https://cplusplus.com/reference/cstdio/fgetc/

Reply:
Yes I agree, I mentioned it :stuck_out_tongue: 

But it looks like our implementation is maybe wrong, as it _internally_ uses `readByte()` which tells to use `eos()` and not check against `EOF`, but the code does the latter.

So is there is a deeper issue in osfgetc() ?

Reply:
Yes, there is a deeper issue here

`osfgetc()` *should* return EOF if eos() is true... so it should be something like this:

    int osfgetc(osfildef *fp) {
        Common::ReadStream *s = dynamic_cast<Common::ReadStream *>(fp);
        return s->eos() ? EOF : s->readByte();
    }